src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 1) {
+      return this[0].toString();
+    }
+
+    let thisButStr = '';
```

Comment:
give this variable more obvious name like `string` or `result`

Reply:
but everything can be string or result and from name thisButStr its obvious that it is this but string

Reply:
in this case you have only one variable which gonna return a result, so that's would be correct to name it `result `to make it obvious

Reply:
nuok